sztp-agent/pkg/dhcp/dhcp_lease.go

```diff
@@ -0,0 +1,33 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+Copyright (C) 2022-2023 Intel Corporation
+Copyright (c) 2022 Dell Inc, or its subsidiaries.
+Copyright (C) 2022 Red Hat.
+*/
+
+// Package dhcp implements the DHCP client
+package dhcp
+
+import (
+	"errors"
+	"log"
+	"os"
+)
+
+const sztpRedirectURL = "sztp-redirect-urls"
+
+// GetBootstrapURLViaLeaseFile returns the sztp redirect URL via DHCP lease file
+func GetBootstrapURLViaLeaseFile(dhcpLeaseFile string) (string, error) {
+	var line string
+	if _, err := os.Stat(dhcpLeaseFile); err == nil {
+		for {
+			line = LinesInFileContains(dhcpLeaseFile, sztpRedirectURL)
+			if line != "" {
+				break
+			}
+		}
+		return ExtractfromLine(line, `(?m)[^"]*`, 1), nil
```

Comment:
see comment above, instead of `return` let's append to a list and return the list at the end...